improver/generate_ancillaries/generate_derived_solar_fields.py

```diff
@@ -167,4 +330,32 @@ def process(
             target_grid, surface_altitude, linke_turbidity
         )
 
-        pass
+        # Altitude specifier is used for cf-like naming of output variable
+        if np.allclose(surface_altitude.data, 0.0):
+            at_mean_sea_level = True
+        else:
+            at_mean_sea_level = False
+
+        irradiance_times = self._get_irradiance_times(
+            time, accumulation_period, temporal_spacing
+        )
+
+        irradiance_data = self._calc_clearsky_irradiance_data(
+            target_grid, irradiance_times, surface_altitude.data, linke_turbidity.data
+        )
+
+        # integrate the irradiance data along the time dimension to get the
+        # accumulated solar irradiance.
+        solar_radiation_data = np.trapz(
```

Comment:
I think moving it into `_calc_clearsky_irradiance_data` makes sense (and renaming to `_calc_clearsky_solar_radiation_data`; this will mean that the actual calculation of this parameter is all being done in one place.